RCTAppleHealthKit/RCTAppleHealthKit.m

```diff
@@ -329,4 +353,99 @@ - (void)getModuleInfo:(NSDictionary *)input callback:(RCTResponseSenderBlock)cal
+    HKQuantity *sugarQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:sugar];
+    HKQuantitySample *sugarSample = [HKQuantitySample quantitySampleWithType:sugarType quantity:sugarQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *carbsType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCarbohydrates];
+    HKQuantity *carbsQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:carbs];
+    HKQuantitySample *carbsSample = [HKQuantitySample quantitySampleWithType:carbsType quantity:carbsQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *proteinType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryProtein];
+    HKQuantity *proteinQuantity = [HKQuantity quantityWithUnit:[HKUnit gramUnit] doubleValue:protein];
+    HKQuantitySample *proteinSample = [HKQuantitySample quantitySampleWithType:proteinType quantity:proteinQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    HKQuantityType *energyType = [HKQuantityType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryEnergyConsumed];
+    HKQuantity *energyQuantity = [HKQuantity quantityWithUnit:[HKUnit kilocalorieUnit] doubleValue:energy];
+    HKQuantitySample *energySample = [HKQuantitySample quantitySampleWithType:energyType quantity:energyQuantity startDate:mealDate endDate:mealDate metadata:metadata];
+    
+    NSArray *mealSamples = @[fatSample, fiberSample, sugarSample, carbsSample, proteinSample, energySample];
+    
+    [self.healthStore saveObjects:mealSamples withCompletion:^(BOOL success, NSError * _Nullable error) {
+        if (success) {
+            resolve([NSNull null]);
+        } else {
+            reject(@"healthkit_error", @"failed to save", error);
+        }
+    }];
+}
+
+-(void)deleteHealthKitMeal:(NSString *)mealId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject
+{
+    NSArray<HKQuantityTypeIdentifier> *identifiers = @[
+                                                       HKQuantityTypeIdentifierDietarySugar,
+                                                       HKQuantityTypeIdentifierDietaryProtein,
+                                                       HKQuantityTypeIdentifierDietaryFatTotal,
+                                                       HKQuantityTypeIdentifierDietaryCarbohydrates,
+                                                       HKQuantityTypeIdentifierDietaryFiber,
+                                                       HKQuantityTypeIdentifierDietaryEnergyConsumed
+                                                       ];
+    
+    NSError *failure;
+    for (HKQuantityTypeIdentifier identifier in identifiers) {
+        HKQuantityType *type = [HKQuantityType quantityTypeForIdentifier:identifier];
+        NSPredicate *predicate = [HKQuery predicateForObjectsWithMetadataKey:@"mealId"];
```

Comment:
We need to use the same API as is in the swift version that we're replacing. This predicate is finding any object in the HK database which has `mealId` as a key. We also want to specify what the `mealId` should equal.

We can use:
```
+ (NSPredicate *)predicateForObjectsWithMetadataKey:(NSString *)key allowedValues:(NSArray *)allowedValues;
```

https://developer.apple.com/documentation/healthkit/hkquery/1614780-predicateforobjectswithmetadatak?language=objc